plugins/pagerduty-incidents/src/components/InstructionsModal.tsx

```diff
@@ -0,0 +1,272 @@
+import type React from "react";
+import { useState, useEffect, useCallback, useMemo } from "react";
+
+import {
+  Box,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  Button,
+  FormControl,
+  FormLabel,
+  Input,
+  FormErrorMessage,
+  Flex,
+} from "@chakra-ui/react";
+
+import { usePluginContext } from "@cortexapps/plugin-core/components";
+
+import { useErrorToast } from "../hooks/uiHooks";
+import { useIsPagerDutyTokenValid } from "../hooks/pagerDutyHooks";
+
+interface InstructionsModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onConfigCompleted: () => void;
+}
+
+const InstructionsModal: React.FC<InstructionsModalProps> = ({
+  isOpen,
+  onClose,
+  onConfigCompleted,
+}) => {
+  const { apiBaseUrl } = usePluginContext();
+
+  const apiOrigin = useMemo(
+    () => (apiBaseUrl ? new URL(apiBaseUrl).origin : ""),
+    [apiBaseUrl]
+  );
+
+  const internalBaseUrl = `${apiOrigin}/api/internal/v1`;
+
+  const [tokenInput, setTokenInput] = useState("");
+  const [debouncedTokenInput, setDebouncedTokenInput] = useState("");
+
+  const tokenIsValid = useIsPagerDutyTokenValid(debouncedTokenInput);
+  const errorToast = useErrorToast();
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    setTokenInput(e.target.value);
+  };
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedTokenInput(tokenInput);
+    }, 100);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [tokenInput]);
+
+  const doAddSecret = useCallback(async (): Promise<boolean> => {
+    console.log("Adding secret");
```

Comment:
Do we want to keep these logging statements in for final ship?

Reply:
we do not lol